lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(attributes)
+    @name = attributes[:name]
+    @address = attributes[:address]
+    @phone = attributes[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
```

Comment:
In this method, if you had a `plate_type` method in your `Vehicle` class, you could set the plate types there and get rid of the lines that are assigning the types in this method. 
You have `@plate_type` in your `initialize` method in your `Vehicle` class but having it as an instance variable there doesn't make it very dynamic and now puts the responsibility on this method to reassign them.